akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1233,9 +1298,84 @@ object NoLogging extends LoggingAdapter {
   final override def isInfoEnabled = false
   final override def isDebugEnabled = false
 
-  final protected override def notifyError(message: String): Unit = ()
-  final protected override def notifyError(cause: Throwable, message: String): Unit = ()
-  final protected override def notifyWarning(message: String): Unit = ()
-  final protected override def notifyInfo(message: String): Unit = ()
-  final protected override def notifyDebug(message: String): Unit = ()
+  final protected[akka] override def notifyError(message: String): Unit = ()
+  final protected[akka] override def notifyError(cause: Throwable, message: String): Unit = ()
+  final protected[akka] override def notifyWarning(message: String): Unit = ()
+  final protected[akka] override def notifyInfo(message: String): Unit = ()
+  final protected[akka] override def notifyDebug(message: String): Unit = ()
+}
+
+class LoggingAdapterExt(delegate: LoggingAdapter) extends LoggingAdapter {
```

Comment:
Looks as if we've traded allocating the array into allocating the wrapper for each call - not a huge win as initially planned.

Would it be possible to make it an `implicit class` and `extends AnyVal` to avoid the allocations?